Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -108,6 +108,12 @@ public double MonoisotopicMass
                 }
                 return (double)ClassExtensions.RoundedDouble(_monoisotopicMass.Value);
             }
+            
+        }
+
+        public void SetMonoisotopicMass(double mass)
+        {
+            _monoisotopicMass = mass;
```

Comment:
seems kind of problematic because the correct mass can't be reconstructed from the sequence alone, you would need protease info. might be better to add a modification to the C term rather than an "invisible" mass shift? it also might be easier to implement